packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -456,6 +456,27 @@ class ThreadChannel extends BaseChannel {
   setAppliedTags(appliedTags, reason) {
     return this.edit({ appliedTags, reason });
   }
+  
+  /**
+   * Pins this thread channel from the forum channel.
+   * @param {string} [reason] Reason for pinning
+   * @returns {Promise<ThreadChannel>}
+   */
+   pin(reason) {
+    if (this.parent.type !== ChannelType.ForumChannel) throw new DiscordjsError(ErrorCodes.CannotPinThreadOutOfForumChannel);
+    return this.edit({ flags: [ChannelFlags.Pinned], reason });
```

Comment:
My bad on this one since I gave you this line - this would remove `ChannelFlags.RequireTag` if it were currently set (and any other future tags. Should do something similar to how you handle remove.